### DISCUSSION TITLE
org.apache.kafka.common.errors.UnsupportedVersionException: Creating topics with default partitions/replication factor are only supported in CreateTopicRequest version 4+.

### DISCUSSION THREAD

**Discussion by hariraogotit** [spring-projects/spring-kafka#2675] [Q&A] [answered]
Team,

  I am using Kafka 2.13 and I get the below error while creating the topic.

org.apache.kafka.common.errors.UnsupportedVersionException: Creating topics with default partitions/replication factor are only supported in CreateTopicRequest version 4+.

I went through https://stackoverflow.com/questions/67513239/failed-to-create-topics-exception-norg-apache-kafka-common-errors-unsupport and understood that the replication factor should be 1 or more in lieu of -1

In my case I do not use Spring-cloud stream.  I am just using Kafka (Producer/Consumer). So kindly advise on which parameter should  I change? 

Any leads are much appreciated. Thanks

**Comment by garyrussell** [comment]
There is no such version as 2.13 - that looks like the scala version (that the broker is compiled against).

This repo is for Spring related discussions; if you are using `kafka-clients` directly, you should address the wider Kafka audience, e.g. on Stack Overflow, or on their mailing lists.

That said, it sounds like your broker is < 2.4 and you are not setting a replication factor when creating topics.

You need to show your code/configuration.

**Reply by hariraogotit** [reply]
@garyrussell Sorry for the confusion

I am using spring-kafka 3.0.5 

**Let me  give more context** 

I am using non blocking retry with exponential back off that creates topics with delay suffixed dynamically.

When it tries to create topics automatically I get 

org.apache.kafka.common.errors.UnsupportedVersionException: Creating topics with default partitions/replication factor are only supported in CreateTopicRequest version 4+

Although it is functionally not breaking anything but would like to stop this error being logged.

I do not have anything related to topic or topic creation in config (application.yml or bean)

Kindly suggest the configuration that is needed to get rid of the error being logged

**Reply by hariraogotit** [reply, marked as answer]
I think I need to use autoCreateTopicsWith in RetryTopicConfigurationBuilder. 

Let me give it a try.

**Reply by garyrussell** [reply]
Right - see its javadocs

```java
	/**
	 * Configure the topic creation behavior to auto create topics with the provided
	 * properties.
	 * @param numPartitions the number of partitions.
	 * @param replicationFactor the replication factor (-1 to use the broker default if the
	 * broker is version 2.4 or later).
	 * @return the builder.
	 */
```
The default is -1.

That said, you should consider upgrading your broker, since it is quite old.
